harness/determined/pytorch/deepspeed/_deepspeed_context.py

```diff
@@ -65,15 +66,57 @@ class DeepSpeedTrialContext(det.TrialContext, pytorch._PyTorchReducerContext):
     5. Disable automatic gradient aggregation for non-pipeline-parallel training.
     """
 
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
-        det.TrialContext.__init__(self, *args, **kwargs)
+    def __init__(
+        self,
+        core_context: det.core.Context,
+        trial_seed: Optional[int],
+        hparams: Optional[Dict],
+        slots_per_trial: int,
+        num_gpus: int,
+        exp_conf: Optional[Dict[str, Any]],
+        aggregation_frequency: int,
+        steps_completed: int,
+        debug_enabled: bool,
+        enable_tensorboard_logging: bool = True,
+    ) -> None:
+        self._core = core_context
+        self.distributed = self._core.distributed
+
         pytorch._PyTorchReducerContext.__init__(self, self.distributed.allgather)
 
+        self._per_slot_batch_size, self._global_batch_size = (
+            util.calculate_batch_sizes(
+                hparams=hparams,
+                slots_per_trial=slots_per_trial,
+                trialname="PyTorchTrial",
+            )
+            if hparams and hparams.get("global_batch_size", None)
+            else (None, None)
+        )
+        self._hparams = hparams
+        self._num_gpus = num_gpus
+        self._debug_enabled = debug_enabled
+        self._exp_conf = exp_conf
+
+        self._trial_seed = trial_seed
+        self._steps_completed = steps_completed
+
         self._init_device()
 
         # Track which types we have issued warnings for in to_device().
         self._to_device_warned_types = set()  # type: Set[Type]
 
+        if aggregation_frequency > 1:
```

Comment:
there's no point is making this argument available if we're just going to say "do it in deepspeed instead"

Reply:
you resolved this but nothing has changed? 👀

Reply:
nvm i was looking at an old commit lol